mypy/checker.py

```diff
@@ -1916,10 +2045,12 @@ def visit_yield_expr(self, e: YieldExpr) -> Type:
     def check_subtype(self, subtype: Type, supertype: Type, context: Context,
                       msg: str = messages.INCOMPATIBLE_TYPES,
                       subtype_label: str = None,
-                      supertype_label: str = None) -> None:
+                      supertype_label: str = None) -> bool:
```

Comment:
But it's not just a fail() call -- I want all the logic below that formats the nice error message. I suppose I could just leave the check_subtype() call out from get_awaitable_return_type() -- then instead of
```
error: Incompatible types in await (actual type Iterator[Any], expected type "Awaitable")
```
you'd get
```
error: Iterator[Any] has no attribute "__await__"
```
(and similar for other variants -- there are four different messages possible, for `await`, `async for`, and two for `async with` (for `__aenter__` and `__aexit__`).

But I prefer to keep it this way -- I'm not sure what's so unidiomatic about my call.